pkg/importer/gcs-datasource.go

```diff
@@ -0,0 +1,107 @@
+package importer
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+
+	"kubevirt.io/containerized-data-importer/pkg/util"
+)
+
+type GCSDataSource struct {
+	// GCS endpoint
+	ep *url.URL
+	// Stack of readers
+	readers *FormatReaders
+	// Reader
+	gcsReader *storage.Reader
+	// The image file in scratch space
+	url *url.URL
+}
+
+func NewGCSDataSource(endpoint string) (*GCSDataSource, error) {
```

Comment:
I'm not familiar with GCS, but are any auth methods that we may want to support?  Looks like only supporting signed or public access URLs?

Reply:
Hi, this implementation works with Google ADC and expects that the ADC is already set up on the worker node. The Google cloud client library will be responsible for looking for the credentials for Google cloud and authenticates to google cloud services (i.e., GCS in this case). So it does support both public and private GCS resources.

As the next step, we are planning to enable CDI to use a different Google service account other than the default one configured on the node to access GCS. The way to configure a different service account is under discussion.